project1/application.py

```diff
@@ -1,75 +1,92 @@
-app.config["SESSION_TYPE"] = "filesystem"
-Session(app)
+APP.config["SESSION_PERMANENT"] = False
+APP.config["SESSION_TYPE"] = "filesystem"
+Session(APP)
 
 # Set up database
-engine = create_engine("postgres://onucnnqeicevfn:c2b89ff6bebb4f290d6b7d5cc6e1ac1ff55b1cee18da41f3145983e36f0cc838@ec2-54-197-48-79.compute-1.amazonaws.com:5432/d8aehqfbjgde2h")
-Base.metadata.create_all(engine)
-db = scoped_session(sessionmaker(bind=engine))
-session=db()
+ENGINE = create_engine(os.getenv("DATABASE_URL"))
+BASE.metadata.create_all(ENGINE)
+DB = scoped_session(sessionmaker(bind=ENGINE))
+DB_SESSION = DB()
 
-@app.route("/")
+@APP.route("/")
 def index():
     return "Project 1: TODO"
 
 
-@app.route("/register",methods = ['POST','GET'])
+@APP.route("/register", methods=['POST', 'GET'])
 def register():
     if request.method == 'POST':
-       name = request.form['uname']
-       print(name)
-       password = request.form['password']
-       
-       r=User(username=name,created_time=datetime.utcnow())
-       r.hash_password(password)
-       try:
-       	   print("saved record")
-       	   if session.query(exists().where(User.username == name)).scalar():
-       	      return render_template('result.html',value="Already registered")
-           else:
-           		db.add(r)
-           		db.commit()
-           		return render_template('result.html',value="successfully registered")
-       except Exception as e:
-           print(e)
-           return render_template('error.html')
+        name = request.form['uname']
+        print(name)
+        password = request.form['password']
+        print(password)
+        record = User(username=name, password=password, created_time=datetime.utcnow())
+      # r.hash_password(password)
+        try:
+            print("saved record")
+            if DB_SESSION.query(exists().where(User.username == name)).scalar():
+                return render_template('result.html', value="Already registered")
+            else:
+                DB_SESSION.add(record)
+                DB_SESSION.commit()
+                return render_template('result.html', value="successfully registered")
+        except Exception as e:
+            print(e)
+            return render_template('error.html')
     return render_template('registration.html')
 
-@app.route("/admin")
+@APP.route("/admin")
 def admin():
- 	res = db.query(User)
- 	print(res)
- 	return render_template('admin.html',result = res)
+    res = DB_SESSION.query(User)
+    print(res)
+    return render_template('admin.html', result=res)
+
+@APP.route('/auth', methods=["POST"])
+def login():
+    data = request.form
+    username = data.get("uname")
+    password = data.get("password")
+   # pwd=generate_password_hash(password)
+    print("username:", username)
+    print("password:", password)
+    users = DB_SESSION.query(User).filter(User.username == username, User.password == password)
+    length = 0
+    for user in users:
+        length = length+1
+
```

Comment:
users is a list and you can use len(users) to check number of users in the list.